python_modules/dagster/dagster/core/log_manager.py

```diff
@@ -148,12 +148,51 @@ def construct_log_string(
     )
 
 
+class PythonLogCaptureHandler(logging.Handler):
+    def __init__(self, level: int, log_manager: "DagsterLogManager"):
+        self._log_manager = log_manager
+        super().__init__(level=level)
+
+    def _extract_extra(self, record: logging.LogRecord) -> Dict[str, Any]:
+        # In the logging.Logger log() implementation, the elements of the `extra` dictionary
+        # argument are smashed into the __dict__ of the underlying logging.LogRecord.
+        # This function figures out what the original `extra` values of the log call were by
+        # comparing the set of attributes in the received record to those of a default record.
+        ref_attrs = list(logging.makeLogRecord({}).__dict__.keys()) + ["message", "asctime"]
+        return {k: v for k, v in record.__dict__.items() if k not in ref_attrs}
+
+    def emit(self, record: logging.LogRecord):
+        # avoid processing record that already have dagster meta
```

Comment:
nit: grammar

Reply:
Agree that this is weird, but I think that this weirdness is just inherited from the weirdness of @logger. In a world where there were only handlers, the two responsibilities would be very distinct: DagsterLogManager is a log that produces messages with Dagster metadata and has a set of handlers on it. DagsterLogConversionHandler is a special handler that sneakily converts normal python logs into Dagster-formatted logs, and then calls all of DagsterLogManager's handlers with this new record.

Things only get tricky because we need to handle the case of taking a record and `emit`ing it, and also taking a record and `log`ing it. However, I can refactor this such that the handler doesn't need to have a back-reference to the DagsterLogManager, which might make things a little cleaner.